### PR TITLE
[Layer dialog] Save new forms

### PR DESCRIPTION
**Pull request by parulraheja98**
Refs: #224

### CODE REVIEW

File: web-ng/src/app/services/data-store/data-store.service.ts [hunk 1]
```diff
@@ -77,11 +77,18 @@ export class DataStoreService {
   // TODO: Define return types for methods in this class
   updateProjectLayer(projectId: string, layer: Layer) {
     const { id: layerId, ...layerDoc } = layer;
+    const updLayer = { ...layerDoc, forms: {} };
+    if (layerDoc.forms) {
+      updLayer.forms = layerDoc.forms.toJS();
```
[the remark this answers is not in the document]
Comment (parulraheja98) [reply web-ng/src/app/services/data-store/data-store.service.ts]: Totally agree :)

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts [hunk 2]
```diff
@@ -69,20 +80,45 @@ export class LayerDialogComponent implements OnDestroy {
...
+      }),
     };
+    if (!this.layerForm.value.question) {
+      delete layer['forms'];
+    }
+
     // TODO: Inform user layer was saved
     this.dataStoreService
-      .updateProjectLayer(this.projectId, layer)
+      .updateProject(this.projectId, layer)
```
Comment (gino-m) [inline web-ng/src/app/components/layer-dialog/layer-dialog.component.ts]: Shouldn't this be `updateLayer`?

File: web-ng/src/app/services/data-store/data-store.service.ts [hunk 3]
```diff
@@ -75,8 +75,13 @@ export class DataStoreService {
...
 
   // TODO: Define return types for methods in this class
-  updateProjectLayer(projectId: string, layer: Layer) {
-    const { id: layerId, ...layerDoc } = layer;
+  updateProject(projectId: string, layer: Layer) {
+    const { id: layerId, forms, ...layerDoc } = layer;
+    if (forms && !forms.isEmpty()) {
+      layerDoc['forms'] = forms.toJS();
+    } else {
+      delete layerDoc['forms'];
```
Comment (gino-m) [inline web-ng/src/app/services/data-store/data-store.service.ts]: `forms` will already be extracted from `layerDoc` at like 79, so this else block isn't necessary.

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts (hunk 2 above)
Comment (parulraheja98) [reply web-ng/src/app/components/layer-dialog/layer-dialog.component.ts]: oops!

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts [hunk 4]
```diff
@@ -35,18 +40,24 @@ export class LayerDialogComponent implements OnDestroy {
...
+
   constructor(
     // tslint:disable-next-line:no-any
     @Inject(MAT_DIALOG_DATA) data: any,
     private dialogRef: MatDialogRef<LayerDialogComponent>,
     private projectService: ProjectService,
     private dataStoreService: DataStoreService,
-    private router: Router
+    private router: Router,
+    private fb: FormBuilder
```
Comment (gino-m) [inline web-ng/src/app/components/layer-dialog/layer-dialog.component.ts]: Nit: avoid acronyms - `formBuilder`, not `fb`

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts [hunk 5]
```diff
@@ -69,20 +80,45 @@ export class LayerDialogComponent implements OnDestroy {
...
   }
 
+  getForm(question: string, fieldId: string, formId: string): Form {
+    const form = {
+      id: formId,
+      fields: Map({
+        [fieldId]: {
+          id: fieldId,
+          type: FieldType['TEXT'],
+          required: true,
```
Comment (gino-m) [inline web-ng/src/app/components/layer-dialog/layer-dialog.component.ts]: Default should be `false`

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts [hunk 6]
```diff
@@ -69,20 +80,45 @@ export class LayerDialogComponent implements OnDestroy {
     this.projectId = project.id;
   }
 
+  getForm(question: string, fieldId: string, formId: string): Form {
```
Comment (gino-m) [inline web-ng/src/app/components/layer-dialog/layer-dialog.component.ts]: I assume this function is a temporary function until we replace with real form designer functionality? If so no action needed (yet).

Comment (parulraheja98) [reply web-ng/src/app/components/layer-dialog/layer-dialog.component.ts]: Yes this is just a temporary function for now :)

Comment (gino-m) [reply web-ng/src/app/components/layer-dialog/layer-dialog.component.ts]: Sgtm

**Pull request merged**